commit 14d912b7c2c3f8602f22864bd3a188c32bb38b5c
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Tue Nov 14 11:03:22 2017 +0000

    mcp2515: Use DT-supplied interrupt flags
    
    The MCP2515 datasheet clearly describes a level-triggered interrupt
    pin. Therefore the receiving interrupt controller must also be
    configured for level-triggered operation otherwise there is a danger
    of a missed interrupt condition blocking all subsequent interrupts.
    The ONESHOT flag ensures that the interrupt is masked until the
    threaded interrupt handler exits.
    
    Rather than change the flags globally (they must have worked for at
    least one user), allow the flags to be overridden from Device Tree
    in the event that the device has a DT node.
    
    See: https://github.com/raspberrypi/linux/issues/2175
         https://github.com/raspberrypi/linux/issues/2263
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/arch/arm/boot/dts/overlays/mcp2515-can0-overlay.dts b/arch/arm/boot/dts/overlays/mcp2515-can0-overlay.dts
index c96cdae27fb1..03eb5486fa9c 100755
--- a/arch/arm/boot/dts/overlays/mcp2515-can0-overlay.dts
+++ b/arch/arm/boot/dts/overlays/mcp2515-can0-overlay.dts
@@ -60,7 +60,7 @@
                 pinctrl-0 = <&can0_pins>;
                 spi-max-frequency = <10000000>;
                 interrupt-parent = <&gpio>;
-                interrupts = <25 0x2>;
+                interrupts = <25 8>; /* IRQ_TYPE_LEVEL_LOW */
                 clocks = <&can0_osc>;
             };
         };
diff --git a/arch/arm/boot/dts/overlays/mcp2515-can1-overlay.dts b/arch/arm/boot/dts/overlays/mcp2515-can1-overlay.dts
index 67bd0d9bdaa2..dc773fa3b50c 100644
--- a/arch/arm/boot/dts/overlays/mcp2515-can1-overlay.dts
+++ b/arch/arm/boot/dts/overlays/mcp2515-can1-overlay.dts
@@ -60,7 +60,7 @@
                 pinctrl-0 = <&can1_pins>;
                 spi-max-frequency = <10000000>;
                 interrupt-parent = <&gpio>;
-                interrupts = <25 0x2>;
+                interrupts = <25 8>; /* IRQ_TYPE_LEVEL_LOW */
                 clocks = <&can1_osc>;
             };
         };
diff --git a/drivers/net/can/spi/mcp251x.c b/drivers/net/can/spi/mcp251x.c
index f3f05fea8e1f..6c2d3aba2591 100644
--- a/drivers/net/can/spi/mcp251x.c
+++ b/drivers/net/can/spi/mcp251x.c
@@ -952,6 +952,9 @@ static int mcp251x_open(struct net_device *net)
 	priv->tx_skb = NULL;
 	priv->tx_len = 0;
 
+	if (spi->dev.of_node)
+	    flags = 0;
+
 	ret = request_threaded_irq(spi->irq, NULL, mcp251x_can_ist,
 				   flags | IRQF_ONESHOT, DEVICE_NAME, priv);
 	if (ret) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1355_14d912b7c2c3f8602f22864bd3a188c32bb38b5c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1355_14d912b7c2c3f8602f22864bd3a188c32bb38b5c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
